### PR TITLE
Add sass-lint

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

File: .sass-lint.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  no-important: 0
+  no-color-keywords: 0
+  no-color-literals: 0
+  single-line-per-selector: 0
+  nesting-depth: 0
+  zero-unit: 0
+  hex-notation:
+    - 2
+    -
+      style: lowercase
```
Comment (danni) [inline .sass-lint.yml]: Move onto same like as hyphen.

File: package.json [hunk 2]
```diff
@@ -52,6 +52,7 @@
     "node-libs-browser": "0.5.2",
     "nodemon": "1.4.1",
     "optimist": "0.6.0",
+    "sass-lint": "^1.2.0",
```
Comment (danni) [inline package.json]: Remove caret.

File: script/generate-component-index [hunk 3]
```diff
@@ -50,14 +50,14 @@ echo "};" >> $target
...
+echo "@import './normalize';" >> $target
+echo "@import './colors';" >> $target
+echo "@import './responsive';" >> $target
 
 echo "" >> $target
 
 for f in $( find src -name '*.scss' -not -path 'src/styles/*.scss' | sort )
 do
-    import=$( echo "$f" | sed s/^src\\//..\\// )
+    import=$( echo "$f" | sed s/^src\\//..\\// | sed s/\.scss$// )
```
Comment (danni) [inline script/generate-component-index]: You can chain regexes using ;

Comment (danni) [inline script/generate-component-index]: Also you don't need to use / as your regex character.

File: src/components/AppBar.scss [hunk 4]
```diff
@@ -31,7 +31,7 @@
         &:hover, &:focus {
 
             svg {
-                background-color: fade-out(white, 0.8) !important;
+                background-color: fade-out(white, .8) !important;
```
Comment (danni) [inline src/components/AppBar.scss]: Removing the leading 0 is a horrid rule :-P

**Comment by danni** [comment]
Is there a rule to enforce order of styles? Ala [Idiomatic CSS](https://github.com/necolas/idiomatic-css)

**Comment by DanielHeath** [comment]
order: smacss

File: .sass-lint.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+#########################
+## Sample Sass Lint File
+#########################
```
Comment (danni) [inline .sass-lint.yml]: Not a sample any more.

File: src/icons/icons.scss [hunk 6]
```diff
@@ -61,7 +61,12 @@
     animation: rotation 3s infinite linear;
 
     @keyframes rotation {
-        from {transform: rotate(0deg);}
-        to   {transform: rotate(359deg);}
+        from {
+            transform: rotate(0deg);
+        }
+
+        to   {
```
Comment (danni) [inline src/icons/icons.scss:68]: Weird spaces now.

**Comment by danni** [comment]
Generally ++

**Pull request merged**